src/views/Peers.tsx

```diff
@@ -71,22 +75,28 @@ export const Peers = () => {
     const actionsMenu = (<Menu items={itemsMenuAction} ></Menu>)
 
     const transformDataTable = (d:Peer[]):PeerDataTable[] => {
-        return d.map(p => ({ key: p.IP, ...p } as PeerDataTable))
+        const peer_ids = d.map(_p => _p.ID)
+        return d.map((p) => {
+            const gs = groups
+                .filter(g => g.Peers?.find((_p:GroupPeer) => _p.ID === p.ID))
+                .map(g => ({ID: g.ID, Name: g.Name, PeersCount: g.Peers?.length, Peers: g.Peers || []}))
+            return {
+                key: p.ID,
+                ...p,
+                groups: gs,
+                groupsCount: gs.length
+            } as PeerDataTable
+        })
     }
 
     useEffect(() => {
         dispatch(peerActions.getPeers.request({getAccessTokenSilently, payload: null}));
         dispatch(groupActions.getGroups.request({getAccessTokenSilently, payload: null}));
-        // dispatch(groupActions.saveGroup.request({getAccessTokenSilently, payload: {
-        //     ID: "caakdnhvdm4s73ak8cgg",
-        //     Name: "ZZZZ",
-        //     Peers: ["wksAVnZSc/ewyU8f8UFqQjjb3TKOqgxSVa0FtDz0jHs="]
-        // } as Group}))
     }, [])
 
     useEffect(() => {
         setDataTable(transformDataTable(peers))
-    }, [peers])
+    }, [peers, groups])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'transformDataTable'. Either include it or remove the dependency array.](https://app.codacy.com/gh/netbirdio/dashboard/pullRequest?prid=9560373)
```suggestion
    }, [peers, groups, transformDataTable])
```